### PR TITLE
Split Android section between WebExtensions and Legacy, added 'differ…

### PR DESCRIPTION
**Pull request by rebloor**
Split Android section between WebExtensions and Legacy, added 'differences' doc to WebExtensions section

### CODE REVIEW

File: macros/AddonSidebar.ejs [hunk 1]
```diff
@@ -117,13 +117,28 @@ function currentPageIsUnder(root) {
...
+
+        <li><a href="<%=baseURL%>Firefox_for_Android/API">APIs</a>
+        <%-
+          wiki.tree(baseURL + "Firefox_for_Android/API", 1, 0, 0, 1)
+        %>
+        </li>
+      </ol> 
+    </ol>
 
     <li><a href="<%=baseURL%>Firefox_for_Android/API">APIs</a>
```
Comment (wbamberg) [inline macros/AddonSidebar.ejs]: The API section is duplicated here

Comment (wbamberg) [inline macros/AddonSidebar.ejs]: I think that the sidebar macro doesn't support the things you want to do here. I don't think you can have this many levels of indentation. Something like this works reasonably well as a replacement for https://github.com/mozilla/kumascript/blob/master/macros/AddonSidebar.ejs#L119-L132:

 ```
    <li><a href="<%=baseURL%>Firefox_for_Android"><strong>Firefox for Android</strong></a></li>
    <li><a href="<%=baseURL%>Firefox_for_Android/WebExtensions">WebExtensions</a>
      <ol>
        <li><a href="<%=baseURL%>Firefox_for_Android/WebExtensions/Differences_between_desktop_and_Android">Differences between desktop and Android</a></li>
      </ol>
    </li>
    <li><a href="<%=baseURL%>Firefox_for_Android/WebExtensions">Legacy add-ons</a>
      <ol>
          <li><a href="<%=baseURL%>Firefox_for_Android/Walkthrough">Walkthrough</a></li>
          <li><a href="<%=locale%>/docs/Tools/Remote_Debugging/Debugging_Firefox_for_Android_with_WebIDE">Debugging</a></li>
          <li><a href="<%=baseURL%>Firefox_for_Android/Code_snippets">Code snippets</a></li>
          <li><a href="<%=baseURL%>Firefox_for_Android/API">APIs</a></li>
      </ol>
    </li>
```

Having said that, I much prefer your suggestion of moving the SDK and the old Firefox for Android docs into "Legacy", and moving the new Firefox for Android docs under "WebExtensions".

Comment (wbamberg) [inline macros/AddonSidebar.ejs]: I would ask Amy if she agrees with this plan.

Comment (rebloor) [inline macros/AddonSidebar.ejs]: Will, I think Amy has this covered for discussion in a [Trello card](https://trello.com/c/KHaM3kcb).

**Comment by rebloor** [comment]
We will now look to reorganizing the sidebar to place legacy content (inc the Firefox for Android section) in a new subsection and add this article to the WebExtensions section

**Pull request closed without merging**